lib/Mongo/MongoDB.php

```diff
@@ -405,7 +405,12 @@ public function command(array $data, $options = [], &$hash = null)
     {
         try {
             $cursor = new \MongoCommandCursor($this->connection, $this->name, $data);
-            $cursor->setReadPreference($this->getReadPreference());
+
+            if (!empty($data['new']) || !empty($data['upsert'])) {
```

Comment:
> Note that there are many more commands that write than findAndModify, which it looks like this is attempting to detect.

Yes, because of this I look values of keys "new" and "upsert". So it works for all commands.